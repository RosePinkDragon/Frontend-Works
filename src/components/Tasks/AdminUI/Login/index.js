import React from "react";
import "styles/login.css";

const LoginScreen = ({ logger }) => {
  return (
    <div className="login">
      <form onSubmit={() => logger(true)}>
        <div className="formElement">
          <label htmlFor="email">Email</label>
          <input type="email" name="email" id="email" required />
        </div>
        <div className="formElement">
          <label htmlFor="password">Password</label>
          <input type="password" name="password" id="password" required />
        </div>
        <button className="btn">login</button>
      </form>
    </div>
  );
};

export default LoginScreen;
