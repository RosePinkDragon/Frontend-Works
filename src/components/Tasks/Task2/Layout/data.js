export const sidebarData = [
  {
    name: "Admin Menu",
    options: ["All Admins", "This Admin", "None"],
  },
  {
    name: "User Menu",
    options: ["All Users", "This User", "None"],
  },
  {
    name: "Client Menu",
    options: ["All Clients", "This Client", "None"],
  },
  {
    name: "Dev Menu",
    options: ["All Devs", "This Dev", "None"],
  },
];
