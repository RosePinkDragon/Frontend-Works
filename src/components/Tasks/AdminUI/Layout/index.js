import React, { useState } from "react";
import userLogo from "images/userLogo.png";
import "styles/Layout.css";
import { sidebarData } from "./data";
import SidebarSubMenu from "./SidebarSubMenu";

export default function Layout({ children, logger }) {
  const [isActive, setIsActive] = useState(false);

  return (
    <>
      <header>
        <nav>
          <h1>App Name</h1>
          <div>
            <img className="logo" src={userLogo} alt="" />
            <p>Moheb Dabilkar</p>
          </div>
          <div
            className={`burger-btn ${isActive ? "active" : ""}`}
            onClick={() => setIsActive(!isActive)}
          />
        </nav>
      </header>
      <div className={`sidebar ${isActive ? "active" : ""}`}>
        <div className="sidebarUserInfo">
          <img className="logo" src={userLogo} alt="" />
          <p>Moheb Dabilkar</p>
        </div>
        <div className="components">
          {sidebarData.map((data, idx) => (
            <SidebarSubMenu key={idx} data={data} />
          ))}
        </div>
        <div>
          <button className="btn logOut" onClick={() => logger(false)}>
            LogOut
          </button>
        </div>
      </div>
      <div className="bodyWrap">{children}</div>
    </>
  );
}
