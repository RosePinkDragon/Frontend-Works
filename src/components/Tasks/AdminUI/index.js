import { useState } from "react";
import Layout from "components/Tasks/AdminUI/Layout";
import LoginScreen from "components/Tasks/AdminUI/Login/";
import { productData } from "components/Tasks/AdminUI/Home/data.js";
import "styles/App.css";
import Chart from "./Home/components/Chart";

// ?? data is used to store external data to reuse.

function AdminUI() {
  const [isLogged, setIsLogged] = useState(true);

  return (
    <>
      {isLogged ? (
        <Layout logger={setIsLogged}>
          <div className="cardWrap">
            {productData.map(({ name, products }) => (
              <div className="card" key={name}>
                <h3>{name}</h3>
                <h5>{products.length}</h5>
              </div>
            ))}
          </div>
          <Chart />
        </Layout>
      ) : (
        <LoginScreen logger={setIsLogged} />
      )}
    </>
  );
}

export default AdminUI;
