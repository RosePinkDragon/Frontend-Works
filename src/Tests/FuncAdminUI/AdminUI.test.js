import React from "react";
import AdminUI from "components/Tasks/AdminUI/";
// import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { productData } from "components/Tasks/AdminUI/Home/data.js";

let wrapped = shallow(<AdminUI />);

describe("Renders the Layout with Children", () => {
  test("Snapshot Renders", () => {
    expect(wrapped).toMatchSnapshot();
  });

  it("should render the login screen", () => {
    wrapped.setState({ isLogged: false });
    expect(wrapped).toMatchSnapshot();
  });

  // let wrapper = renderer.create(<Layout>{layoutChildren}</Layout>);
  //   const childWrapper = wrapper.find(".bodyWrap");
  expect(wrapped.find(".card").length).toEqual(productData.length);
});
