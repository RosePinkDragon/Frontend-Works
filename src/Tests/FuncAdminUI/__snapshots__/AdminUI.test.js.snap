// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Layout with Children Snapshot Renders 1`] = `
<Fragment>
  <Layout
    logger={[Function]}
  >
    <div
      className="cardWrap"
    >
      <div
        className="card"
        key="Passed Products"
      >
        <h3>
          Passed Products
        </h3>
        <h5>
          6
        </h5>
      </div>
      <div
        className="card"
        key="Upcoming Products"
      >
        <h3>
          Upcoming Products
        </h3>
        <h5>
          6
        </h5>
      </div>
      <div
        className="card"
        key="Total Products"
      >
        <h3>
          Total Products
        </h3>
        <h5>
          6
        </h5>
      </div>
    </div>
    <Chart />
  </Layout>
</Fragment>
`;
